base/REPL.jl

```diff
@@ -153,9 +153,9 @@ function print_response(errio::IO, val::ANY, bt, show_value::Bool, have_color::B
                 if val !== nothing && show_value
                     try
                         if specialdisplay === nothing
-                            display(val)
+                            eval(Main, Expr(:call, () -> display(val)))
```

Comment:
There use to be cases where that wouldn't work (such as https://github.com/JuliaLang/julia/issues/15310 or `Expr(:$)`, IIRC). Do you think those are all fixed now?